### ISSUE TITLE
Use better-sqlite3 instead of sqlite3 for performance

### ISSUE THREAD

**Issue by vicb** [mapbox/node-mbtiles#100]
I have found [better-sqlite3](https://www.npmjs.com/package/better-sqlite3) to perform much better than [sqlite3](https://www.npmjs.com/package/sqlite3) on one of my project.

They advertise that speed and also have better docs.

If you are interested in switching, I could work on the PR, let me know.

**Comment by jingsam** [comment]
This looks very promising. I would like to take a look. @springmeyer  @kkaefer  What do you think about it?

**Comment by vicb** [comment]
One issue might be that the extension is compiled at install time meaning
that you should have a dev environment installed where you install the lib.
OTOH I think sqlite3 comes pre-built.
If I remember correctly there is an issue on the better-sqlite3 repo.

**Comment by springmeyer** [comment]
This seems like reasonable thing to consider. To approve the switch I'd want to see a benchmark written that replicates an important usecase of node-mbtiles and shows that the switch concretely improves performance (both increased speed and reduced memory usage). So the benchmark should be able to be run against both node-sqlite3 and better-sqlite3. We could each run that benchmark on a few different machines to ensure that the code is working well with better-sqlite3.

**Comment by springmeyer** [comment]
> OTOH I think sqlite3 comes pre-built.

Yes, node-sqlite3 provides pre-built binaries. Ideally better-sqlite3 would too. But that does not need to be a blocker for writing a benchmark to compare.

**Comment by jefbarn** [comment]
> warning @mapbox/mbtiles > sqlite3 > node-pre-gyp@0.11.0: Please upgrade to @mapbox/node-pre-gyp: the non-scoped node-pre-gyp package is deprecated and only the @mapbox scoped package will recieve updates in the future

Seems like the sqlite version this package is using is kind-of wrapped around the axle. Suggestions?
